spec/ssh_scan/rule_engine_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rspec'
+require 'ssh_scan/rule_engine'
+
+describe SSHScan::RuleEngine do
+  context "for conditionals" do
+    it "eq, neq should work for any literal" do
+      result = {:ssh_lib => "openssh"}
+      expect(SSHScan::RuleEngine.match({"ssh_lib_eq" => "openssh"}, result)).to be_truthy
```

Comment:
recommend be_true here.   In ruby, "true" is not the same as "truthy".

Since your match routine is always true/false, I would recommend constraining your unit-tests to look for those specifically.

Reference: for more truthy education (https://gist.github.com/jfarmer/2647362)